core/src/main/java/io/grpc/internal/GrpcUtil.java

```diff
@@ -82,8 +87,20 @@
   /**
    * {@link io.grpc.Metadata.Key} for the accepted message encodings header.
    */
-  public static final Metadata.Key<String> MESSAGE_ACCEPT_ENCODING_KEY =
-          Metadata.Key.of(GrpcUtil.MESSAGE_ACCEPT_ENCODING, Metadata.ASCII_STRING_MARSHALLER);
+  public static final Metadata.Key<byte[]> MESSAGE_ACCEPT_ENCODING_KEY =
+      InternalMetadata.keyOf(GrpcUtil.MESSAGE_ACCEPT_ENCODING, new AcceptEncodingMarshaller());
+
+  private static final class AcceptEncodingMarshaller implements TrustedAsciiMarshaller<byte[]> {
```

Comment:
Surprisingly (in a likely bad way), I think this is okay.  Existing Ascii marshallers just do the conversion inside the AsciiKey, which seems about as bad as doing it inside of DecompressorRegistry.  The TransportFrameUtil double checks that all the characters are still in range post conversion, so I think it's okay.

Decompressor name is checked for commas when added to the registry, so I *think* it is still safe.  

The decoding path is safe for the same reasons.